### PR TITLE
Fix or disable failing tests

### PR DESCRIPTION
**Pull request by Smjert**
Start working toward getting the Azure Pipelines CI all green.

#5608

### CODE REVIEW

**Comment by Smjert** [comment]
Thanks for the review!

Sorry I noticed that I mixed variable naming styles; when the PR could be merged I'll squash that with its respective commit (https://github.com/osquery/osquery/pull/5614/commits/303d97a4ff113d4f025e5e52a25c4ddaf9a4310a)

**Comment by Smjert** [comment]
I've squashed now, I remembered that it would make reviews stale again if I did it later.

File: tests/integration/tables/interface_details.cpp [hunk 1]
```diff
@@ -26,6 +28,11 @@ class InterfaceDetailsTest : public testing::Test {
 };
 
 TEST_F(InterfaceDetailsTest, test_sanity) {
+#ifdef OSQUERY_WINDOWS
+  LOG(INFO) << "Test failing on Windows, temporarily disabled";
```
Comment (directionless) [inline tests/integration/tables/interface_details.cpp]: I'd wish we had a consistent pattern for this, something to more easily look for. But this seems fine

Comment (Smjert) [reply tests/integration/tables/interface_details.cpp]: We can eventually have a function in some existing test helper code that checks for a given list of platforms and prints a standard message.

Obviously there are a couple of other ways to disable tests, which though either disable entirely a single test case or a test suite.
For googletest this is prepending DISABLED_ to the test case name (for instance `TEST_F(InterfaceDetailsTest, DISABLED_test_sanity`) or test suite name (for instance `class DISABLED_InterfaceDetailsTest : public testing::Test {`)

**Comment by directionless** [comment]
@Smjert I've been assuming you can [technically] merge this with my approval, and will do so when you choose? Or are you waiting on someone?

**Comment by Smjert** [comment]
> @Smjert I've been assuming you can [technically] merge this with my approval, and will do so when you choose? Or are you waiting on someone?

I'll merge it, I was mulling over unifying some other tests changes to this but I already made different PRs.

**Pull request merged**